pkg/resource/interfaces.go

```diff
@@ -137,3 +137,15 @@ type Provider interface {
 
 	CredentialsSecretReferencer
 }
+
+// A Target is a Kubernetes object that refers to credentials to connect
+// to a deployment target. Target is a subset of the Claim interface.
+type Target interface {
+	runtime.Object
+	metav1.Object
+
+	LocalConnectionSecretWriterTo
+	ManagedResourceReferencer
```

Comment:
Doesn't this interface usually correspond to the getters and setters for `resourceRef`? Are we reusing that for `clusterRef` here?

Reply:
Yes. This one is less contentious than the `LocalConnectionSecretWriterTo` because there is nothing that mandates that the `Target` will need to implement this existing interface. However, if we create another interface it will look very similar and function the exact same. I think the reason I did not add a new one is because though we have no plans for it at the moment, I feel that `Target` could be reused to propagate secrets from other types of resources in the future. Do you think we should add a `ClusterReferencer` interface or `ManagedClusterReferencer` interface?

Reply:
I'm a little worried about using getter and setter interfaces that correspond to fields with different names - at the very least it goes against the established convention in Crossplane. I don't feel super strongly about this, but it tingles my spider sense.

Reply:
The secret propagator actually mandates this sub-interface as well, so I have left it as-is and tried to make `Target` a little more compliant by changing the struct field to `ResourceReference` but keeping the JSON tag as `clusterRef`.